store.go

```diff
@@ -1109,6 +1110,14 @@ func (s *store) getROLayerStoresLocked() ([]roLayerStore, error) {
 		return nil, err
 	}
 	for _, store := range s.graphDriver.AdditionalImageStores() {
+		if strings.Contains(store, "|") {
+			parts := strings.Split(store, "|")
+			if len(parts) > 1 && parts[1] == "NO_REUSE" {
+				continue
+			} else {
+				store = strings.Split(store, "|")[0]
+			}
+		}
```

Comment:
I can’t see how this would work; layers in these stores would be effectively invisible, but images in those stores would be visible as usual.

The immediate effect is that all images from a no-reuse store are visible but have unresolvable layers = look broken to all existing tooling.

Reply:
I have tested the changes in 3.X branch, as part of our older software version which manifested the issue and verified basic operations. Admittedly the codebase has changed substantially and the current patch as you described doesn't work anymore.
I will check if there is still a manageable way to introduce the change in main branch.

Reply:
Note use cases like `podman history` and `podman image tree`.